### ISSUE TITLE
[-32023] Could not create account.

### ISSUE THREAD

**Issue by zephyraijunzo** [ethcore/parity#4764]
Hello all,

I have an issue restoring my account from my backup JSON file to Parity.

It shows **[-32023] Could not create account.**

My account shows up fine in the Add New Address area tab but I cannot use it as my main account. In addition, recovering from my recovery phrase, it shows up as an entirely new account and not the original one...

Can anyone help me?

Thanks

**Comment by gavofyork** [comment]
on what platform, and how long ago, did you create your recovery phrase?

what platform are you using now?

how are you attempting to restore the JSON file?

**Comment by zephyraijunzo** [comment]
Hi,

I'm new to this whole scene. I recently format my HDD and installed Window 10 March 3, 2017. Reinstalled Parity 1.5.6 for Windows 64 and run it by following the steps for creating new account since there's no options to restore from backup from the startup. After creating a temp account I proceed to create another new account because there's options to choose:

- create new account manually
- create from recovery phrase
- import accounts from Geth Keystore
- import account from a backup JSON file
- import account from an Ethereum pre-sale wallet
- import ray private key

I chose the JSON back up file and input my username and password as they were before. I get an error code [-32023] as a result. I tried uninstalling and reinstalling to previous versions of Parity and no luck...I tried 

![capture](https://cloud.githubusercontent.com/assets/25127799/23639650/aebec54e-029d-11e7-8443-ade65f3696a9.JPG)

However, in my address book. My accounts are there but I cannot use it as my main account. 

![capture](https://cloud.githubusercontent.com/assets/25127799/23639749/3a6af914-029e-11e7-8794-3e9fe9e2a3bc.JPG)

On top of that I discovered that my recovery phrase was for my other account and not this one so I only have this JSON file...please help and thank you

**Comment by tomusdrw** [comment]
Hi!
Could you manually copy your `json` file to `C:\Users\<yourusername>\AppData\Roaming\Parity\Ethereum\keys\ethereum\` to see if it works?

Where did you get the key file in a first place? Could you open it in Notepad (don't post it here) and see if it says `"encrypted": false` inside.

**Comment by zephyraijunzo** [comment]
Hi,

Could you manually copy your json file to C:\Users\<yourusername>\AppData\Roaming\Parity\Ethereum\keys\ethereum\ to see if it works?

**Yes, I tried multiple times copying my JSON file to that folder but no luck. I even tried changing the naming files to be similar to the others and nothing...**

Where did you get the key file in a first place? Could you open it in Notepad (don't post it here) and see if it says "encrypted": false inside.

**I needed a backup file and I wasn't sure where I would get it but in my account there's an option to export my account. That's where I got my JSON file. I read somewhere that your JSON file is important and it's your backup file when you need to restore your account in Parity. 

Yes, I can open it in Notepad and No, it doesn't say "encrypted" anywhere. I double checked my other accounts and the files look the similar to one another.**

I'm speculating that I didn't input my password right or my username isn't the exact match so Parity cannot restore my account. Because I tried it with my other account and it work but I did it using the recovery phrase. Maybe I should try it using just the password to see if it has the same issue? 

I will keep on trying..!

**Comment by tomusdrw** [comment]
Does the file look like this (when opened in notepad)?
```
{"id":"b88afe96-c33c-f2fc-51f9-cd3ff2b79427","version":3,"crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"<hex>"},"ciphertext":"<hex>","kdf":"pbkdf2","kdfparams":{"c":10240,"dklen":32,"prf":"hmac-sha256","salt":"<hex>"},"mac":"<hex>"},"address":"00e985bd87d244d4616ab9cda0360f2aee6fd584","name":"hat man","meta":"{\"timestamp\":1488961397356}"}
```

Do you get any warning when starting parity about invalid key format?
After you copy the key to the aforementioned folder what does:
```
$ parity account list --chain=homestead
```
return?

**Comment by zephyraijunzo** [comment]
Hi,

Mine doesn't look like what you have. 

Here's is what mine looks like and I have replace the numbers with underlines:

![image](https://cloud.githubusercontent.com/assets/25127799/23732055/baa14c90-0425-11e7-86f6-e36d26ff4931.png)

I didn't get a warning for invalid key format. Just that "[-32023] Could not create account" warning.

I don't understand your last question since I don't know what this is: 
$ parity account list --chain=homestead

Sorry, I'm very new to all of this.

**Comment by tomusdrw** [comment]
So that is not an account backup file. It just contains a list of your accounts, but without any details about private keys. I filled an issue about this #4824.

The best (and only) way to backup a key files currently is to copy all `.json` files from `C:\Users\<username>\AppData\Roaming\Parity\Ethereum\keys\`.

**Comment by tomusdrw** [comment]
Just saw that you also have a recovery phrase. Please try restoring from this (note there is a checkbox to select if you have created this account on specific Parity version on windows).

**Comment by zephyraijunzo** [comment]
Thank you for your inputs tomusdrw!

I looked into my other backup files named "My Ethereum wallet" that I created when I was playing around with Mist and later transferred to Parity.

There are several files I backed up but I only opened the one with my funds and main account. I matched it up with the examples you had and it's a match. 

I proceed to restoring my account using this JSON file in Parity and the result:

![image](https://cloud.githubusercontent.com/assets/25127799/23750711/15c9e5c0-0483-11e7-90e7-6924fe390c7c.png)

What a relief! 

You're correct. The export button is misleading. 

I will backup my files again and make notes. Thank you tomusdrw and gavofyork!
